ui/src/components/Secrets.vue

```diff
@@ -86,43 +86,89 @@
           </v-col>
         </v-row>
 
-        <v-row
-          v-for="(secret, counter) in secrets"
-          :key="counter"
-        >
-          <v-col
-            cols="12"
-            md="4"
-          >
+        <v-row v-for="(secret, counter) in secrets" :key="counter">
+          <v-col cols="12" md="4">
             <v-textarea
               v-model="secret.key"
-              label="Secret key"
+              label="Secret Key"
               rows="1"
               clearable
               prepend-icon="mdi-delete"
               :rules="rules.validSecretKey"
               @click:prepend="removeSecret(counter)"
             />
           </v-col>
-          <v-col
-            cols="12"
-            :sm="hasValue[counter] ? 11 : hasFile[counter] ? 1 : 6"
-            :md="hasValue[counter] ? 7 : hasFile[counter] ? 1 : 4"
-          >
+          <v-col cols="12" :sm="hasValue[counter] ? 11 : hasFile[counter] ? 1 : 6" :md="hasValue[counter] ? 7 : hasFile[counter] ? 1 : 4">
             <v-textarea
               v-model="secret.value"
               rows="1"
               auto-grow
               clearable
-              label="Secret value"
+              label="Secret Value"
               :disabled="hasFile[counter]"
-            />
+            >
+              <template v-slot:prepend>
+                <v-tooltip bottom>
+                  <template v-slot:activator="{ props }">
+                    <v-icon
+                      v-bind="props"
+                      @click="secrets[counter].menuVisible = !secrets[counter].menuVisible"
+                      v-if="enableRandomStringGenerator"
+                    >
+                      mdi-dice-multiple
+                    </v-icon>
+                  </template>
+                  <span>Generate Random String</span>
+                </v-tooltip>
+              </template>
+            </v-textarea>
+            <v-menu
+              v-model="secrets[counter].menuVisible" 
+              :close-on-content-click="false"
+              absolute
+              offset-y
+              v-if="enableRandomStringGenerator"
+            >
+              <template v-slot:activator="{ props }">
+                <v-btn icon v-bind="props" style="display: none;"></v-btn>
+              </template>
+              <v-list>
+                <v-list-item @click.stop>
+                  <v-checkbox v-model="includeUppercase" label="Include Uppercase Letters (A-Z)" />
```

Comment:
This could be then loop over the structure `randomStringGenerator.classes` from the configuration. The label could be read from the description and the v-model could be a hash, where the key is re-used.

That way each backend instance could configure independently, which classes (and chars inside those classes) it would allow.